modal/_utils/async_utils.py

```diff
@@ -463,3 +478,114 @@ def run_generator_sync(
         except BaseException as err:
             exc = err
     loop.close()
+
+
+@asynccontextmanager
+async def aclosing(
```

Comment:
I'd prefer if this *only* takes `AsyncGenerators` and we don't apply it to iterators that don't suffer the same problem. In cases we are not sure if the input is a generator or an non-generator iterator we could have compose `aclosing(ensure_gen(input))` instead